### PR TITLE
KAFKA-8784: remove default close for RocksDBConfigSetter

### PR DESCRIPTION
**Pull request by wcarlson5**
remove default

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

**Comment by wcarlson5** [comment]
@ableegoldman for review

File: streams/src/test/java/org/apache/kafka/test/MockRocksDbConfigSetter.java
Comment (ableegoldman) [inline streams/src/test/java/org/apache/kafka/test/MockRocksDbConfigSetter.java]: I think this mock is probably only used internally by Streams tests, so we don't need to log anything for users here

**Review by ableegoldman — COMMENTED** [review]
Thanks for the quick PR -- one other thing, can you add a note to the streams upgrade guide since this is going to be a breaking change that some users will need to modify their code for?

**Review by ableegoldman — APPROVED** [review]
LGTM, please ping me when the build passes

**Comment by ableegoldman** [comment]
Hey @wcarlson5 I'm having trouble viewing the test failures, the only one I can actually see is `connect.mirror.integration.MirrorConnectorsIntegrationTest.testReplication()` but all three builds are showing failures (JDK15 timed out but that seems to be a known unrelated issue). The only other thing is ` [Checks API] No suitable checks publisher found.` which seems like it's probably environmental.

I assume these are all unrelated but can you just run the tests locally for me to make sure before I merge?

**Comment by wcarlson5** [comment]
@ableegoldman the tests all passed locally

**Pull request merged**
